custom_components/foxess_modbus/entities/charge_period_descriptions.py

```diff
@@ -18,7 +18,7 @@
                     period_end_address=41003,
                     enable_charge_from_grid_address=41001,
                 ),
-                models=Inv.H1_G1 | Inv.KH_PRE119,
+                models=Inv.H1_G1 | Inv.H1_G2 | Inv.KH_PRE119,
```

Comment:
Aha, the G2 uses holding registers here, not input registers. That's why the card isn't loading for you.

Reply:
Confirmed - the charge-period related entities are now created correctly, and the card now loads, and is operational.